@here/harp-mapview/lib/VisibleTileSet.ts

```diff
@@ -1241,30 +1240,25 @@ export class VisibleTileSet {
         const dataSourceCache = this.m_dataSourceCache;
         const retainedTiles: Set<TileCacheId> = new Set();
 
-        function markTileDirty(tile: Tile, tileGeometryManager: TileGeometryManager) {
+        function markTileDirty(tile: Tile) {
             const tileKey = DataSourceCache.getKeyForTile(tile);
             if (!retainedTiles.has(tileKey)) {
                 retainedTiles.add(tileKey);
-                if (tile.tileGeometryLoader !== undefined) {
```

Comment:
doesnt it still need some kind of tile.dispose() call or something similar?

Reply:
I've moved the reset of the tileGeometryLoader to tile.load(). This reset is done to mark a tile as dirty, so we want to keep the tile contents till the new contents are loaded.